libyul/backends/evm/OptimizedEVMCodeTransform.cpp

```diff
@@ -497,7 +497,133 @@ void OptimizedEVMCodeTransform::operator()(CFG::BasicBlock const& _block)
 			CFG::BuiltinCall const* builtinCall = get_if<CFG::BuiltinCall>(&_block.operations.back().operation);
 			yulAssert(builtinCall, "");
 			yulAssert(builtinCall->builtin.get().controlFlowSideEffects.terminatesOrReverts(), "");
-		}
+		},
+		[&](CFG::BasicBlock::Switch const& _switch)
+		{
+			std::cout << "In OptimizedEVMCodeTransform: creating switch statement" << std::endl;
+
+			// Create labels for the targets, if not already present.
+			if (_switch.defaultCase != nullptr && !m_blockLabels.count(_switch.defaultCase))
+				m_blockLabels[_switch.defaultCase] = m_assembly.newLabelId();
+			for (auto const& [caseValue, caseBlock]: _switch.cases)
+				if (!m_blockLabels.count(caseBlock))
+					m_blockLabels[caseBlock] = m_assembly.newLabelId();
```

Comment:
Thinking about it, there is one different option, other than adding the switch expression back to the stack layouts of the switch cases:

Instead of creating the actual block labels here, you could just store "case dispatch" labels in a vector and jump to those, and then below, emit those labels, pop the switch expression and only then visit the case block. That way you could keep the case stack layouts free of the switch expression and we don't need to readjust things for the jump table implementation.

But either way is fine - we'll in any case first need a traditional implementation with switch exits that works reliably before we can move to jump tables.

Reply:
But yeah, that may be nicer - conceptually the case blocks doesn't really *want* the switch expression in its entry layout, so the fact that a traditional switch dispatch needs them is more of an implementation detail of the switch dispatch.